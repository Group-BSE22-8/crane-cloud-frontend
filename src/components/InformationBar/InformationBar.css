.InformationBar {
  background: #f8f7e5;
  height: 4rem;
  display: flex;
  align-items: center;
}

.InformationBarWithButton {
  width: 100%;
  display: flex;
  gap: 1rem;
  justify-content: space-between;
}

.InfoHeader {
  display: flex;
  align-items: center;
  font-weight: bold;
  font-size: 1.5rem;
}

.InfoContent{
  display: flex;
  justify-content: flex-end;
  gap: 4rem;
  align-items: center;
}
.CreditsCorner{
  display: grid;
  align-items: center;
  grid-template-columns: 1fr 1fr;
  color: #F7B21F;
  grid-gap: .5rem;
  width: 3rem;
}

.ViewAppBtn {
  color: #008ac1;
  background-color: transparent;
  outline: 2px solid #008ac1;
}
.ViewAppBtn:hover {
  color: #fff;
  background-color: #008ac1;
  outline: 2px solid #008ac1;
}

.SearchInput{
  display:flex;
}

.SearchInput input{
  background-color: #EBEAD5;
  color: #000;
  border: 0;
  border-top-left-radius: 4px 2px;
  border-bottom-left-radius: 3px 4px;
  font-size: 1rem;
  padding: 1rem;
  height : 45.5px;
  max-width: 21rem;
}

.SearchIcon{
  background-color: #EBEAD5;
  padding: 0.42rem;
  border-bottom-right-radius: 6px 2px;
  border-top-right-radius: 3px 4px;
}



@media only screen and (max-width: 650px) {
  .RoundAddButtonWrap {
    position: absolute;
    top: 9rem;
    right: 1rem;
  }
}
