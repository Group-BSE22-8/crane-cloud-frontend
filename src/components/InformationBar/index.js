import React, { useState } from "react";
//import RoundAddButton from "../RoundAddButton";
import NewButton from "../NewButton";
import AppStatus from "../AppStatus";
import PrimaryButton from "../PrimaryButton";
import { ReactComponent as SearchButton } from "../../assets/images/search.svg";
import { ReactComponent as Coin } from "../../assets/images/coin.svg";
import "./InformationBar.css";

const InformationBar = ({
  header,
  buttontext,
  status,
  showBtn,
  btnAction,
  viewAppLink,
  credits,
  showSearchBar,
  placeholder,
  searchAction,
}) => {
  const [Searchword, setSearchword] = useState("");
  const callbackSearchWord = ({ target }) => {
    const { value } = target;
    setSearchword(value);
    searchAction(value);
  };
  return (
    <div className="InformationBar SmallContainer">
      {status ? (
        <div className="InformationBarWithButton">
          <div className="AppUrl">
            <a target="_blank" rel="noopener noreferrer" href={header}>
              {header}
            </a>
          </div>
          <div className="RoundAddButtonWrap">
            <AppStatus appStatus={status} />
          </div>
        </div>
      ) : showSearchBar ? (
        <div className="InformationBarWithButton">
          <div className="InfoHeader">{header}</div>
          <div className="InfoContent">
            <div className="SearchBar">
              <div className="SearchInput">
                <input
                  type="text"
                  className="searchTerm"
                  name="Searchword"
                  placeholder={placeholder}
                  value={Searchword}
                  onChange={callbackSearchWord}
                />
                <SearchButton className="SearchIcon" />
              </div>
            </div>
            <div className="RoundAddButtonWrap">
              {/*<RoundAddButton onClick={btnAction} />*/}
              <NewButton label={buttontext} type="new" onClick={btnAction} />
            </div>
          </div>
        </div>
      ) : showBtn ? (
        <div className="InformationBarWithButton">
          <div className="InfoHeader">{header}</div>
          <div className="RoundAddButtonWrap">
            {/*<RoundAddButton onClick={btnAction} />*/}
            <NewButton label={buttontext} type="new" onClick={btnAction} />
          </div>
        </div>
      ) : viewAppLink ? (
        <div className="InformationBarWithButton">
          <div className="InfoHeader">{header}</div>
          <a href={viewAppLink} rel="noopener noreferrer" target="_blank">
            <PrimaryButton className="ViewAppBtn">Open App</PrimaryButton>
          </a>
        </div>
      ) : credits ? (
        <div className="InformationBarWithButton">
          <div className="InfoHeader">{header}</div>
          {/**appears if user has credits */}
          <div className="CreditsCorner" title="credits">
            {credits > 0 ? credits : 0}
            <Coin />
          </div>
        </div>
      ) : (
        <div className="InfoHeader">{header}</div>
      )}
    </div>
  );
};

export default InformationBar;
