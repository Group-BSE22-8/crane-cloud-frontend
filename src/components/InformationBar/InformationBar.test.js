import React from "react";
import { shallow } from "enzyme";
import InformationBar from ".";

describe("Test the information bar component", () => {
  const searchAction = jest.fn();
  const btnAction = jest.fn();

  const defaultProps = {
    header: "test",
    status: undefined,
    showBtn: false,
    btnAction: btnAction,
    viewAppLink: undefined,
    showSearchBar: false,
    placeholder: undefined,
    searchAction: searchAction,
  };
  const appStatusProps = {
    ...defaultProps,
    status: "success",
  };
  const searchBarProps = {
    ...defaultProps,
    showSearchBar: true,
    placeholder: "test",
  };
  const addBtnProps = { ...defaultProps, showBtn: true };
  const addAppLinkProps = { ...defaultProps, viewAppLink: "test_url" };

  const InformationBarDefault = shallow(<InformationBar {...defaultProps} />);
  const InformationBarSearch = shallow(<InformationBar {...searchBarProps} />);
  const InformationBarAppLink = shallow(
    <InformationBar {...addAppLinkProps} />
  );
  const InformationBarStatus = shallow(<InformationBar {...appStatusProps} />);
  const InformationBarAddBtn = shallow(<InformationBar {...addBtnProps} />);

  it("checks if the information bar component matches the snapshot", () => {
    expect(InformationBarDefault).toMatchSnapshot();
  });

  it("checks if search in information bar is funtional", () => {
    const SearchInput = InformationBarSearch.find(
      ".DesktopView .SearchInput input"
    );
    SearchInput.simulate("change", {
      target: { name: "Searchword", value: "sample search" },
    });
    expect(searchAction).toBeCalled();
  });
});
