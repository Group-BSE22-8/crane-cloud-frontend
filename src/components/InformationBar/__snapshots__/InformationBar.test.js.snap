// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the information bar component checks if the information bar component matches the snapshot 1`] = `ShallowWrapper {}`;
